test/unit/unit/format.js

```diff
@@ -0,0 +1,295 @@
+    "1 day"
+	);
+	assert.equal(
+    formatUnit( 2, "day", { form: "short" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2 days"
+	);
+	assert.equal(
+    formatUnit( 1, "week", { form: "short" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "1 wk"
+	);
+	assert.equal(
+    formatUnit( 2, "week", { form: "short" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2 wks"
+	);
+	assert.equal(
+    formatUnit( 1, "month", { form: "short" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "1 mth"
+	);
+	assert.equal(
+    formatUnit( 2, "month", { form: "short" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2 mths"
+	);
+	assert.equal(
+    formatUnit( 1, "year", { form: "short" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "1 yr"
+	);
+	assert.equal(
+    formatUnit( 2, "year", { form: "short" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2 yrs"
+	);
+});
+
+QUnit.test( "Narrow form", function( assert ) {
+	assert.equal(
+    formatUnit( 1, "millisecond", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "1ms"
+	);
+	assert.equal(
+    formatUnit( 2, "millisecond", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2ms"
+	);
+	assert.equal(
+    formatUnit( 1, "second", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "1s"
+	);
+	assert.equal(
+    formatUnit( 2, "second", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2s"
+	);
+	assert.equal(
+    formatUnit( 1, "minute", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "1m"
+	);
+	assert.equal(
+    formatUnit( 2, "minute", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2m"
+	);
+	assert.equal(
+    formatUnit( 1, "hour", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "1h"
+	);
+	assert.equal(
+    formatUnit( 2, "hour", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2h"
+	);
+	assert.equal(
+    formatUnit( 1, "day", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "1d"
+	);
+	assert.equal(
+    formatUnit( 2, "day", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2d"
+	);
+	assert.equal(
+    formatUnit( 1, "week", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "1w"
+	);
+	assert.equal(
+    formatUnit( 2, "week", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2w"
+	);
+	assert.equal(
+    formatUnit( 1, "month", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "1m"
+	);
+	assert.equal(
+    formatUnit( 2, "month", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2m"
+	);
+	assert.equal(
+    formatUnit( 1, "year", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "1y"
+	);
+	assert.equal(
+    formatUnit( 2, "year", { form: "narrow" }, oneOrOtherPluralGenerator, cldr, globalize ),
+    "2y"
+	);
+});
```

Comment:
Let's simplify the whole assertions above by creating a custom assertion like https://github.com/jquery/globalize/blob/master/test/unit/date/format.js#L60-L71. This way we avoid boilerplate code.